src/Sylius/Bundle/ThemeBundle/Repository/FailsafeThemeRepository.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ThemeBundle\Repository;
+
+use Sylius\Component\Resource\Model\ResourceInterface;
+
+/**
+ * @author Kamil Kokot <kamil.kokot@lakion.com>
+ */
+final class FailsafeThemeRepository implements ThemeRepositoryInterface
+{
+    /**
+     * @var ThemeRepositoryInterface
+     */
+    private $unstableRepository;
```

Comment:
I don't seem to understand why you need this fail safe repository? It does some real unexpected behaviour, like you think it is added to the db, but it is not and you will not get notified about it.